.github/workflows/main.yml

```
@@ -1,47 +1,125 @@
 on:
-  pull_request:
-    types: [opened, synchronize]
+  workflow_dispatch:
+    inputs:
+      dispatch_token:
+        description: 'Token that authorize the dispatch'
+        required: true
+      head_sha:
+        description: 'Head commit SHA that dispatched the workflow'
+        required: true
+      pr_author_username:
+        description: 'Pull Request author username'
+        required: true
+      pr_number:
+        description: 'Pull Request number that dispatched the workflow'
+        required: true
 
 jobs:
-  eslint:
-    runs-on: ubuntu-18.04
+  ESLint-Evaluator:
+    runs-on: self-hosted
+    name: ESLint
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-node@v1.4.4
+      - name: Fetch project repository
+        uses: actions/checkout@v2
+
+      - name: Fetch Blocked Files Checkout action
+        uses: actions/checkout@v2
         with:
-          node-version: '12'
-      - name: ESLint
-        uses: betrybe/eslint-linter-action@v2.1
+          repository: betrybe/blocked-files-checkout-action
+          ref: v2
+          token: ${{ secrets.GIT_HUB_PAT }}
+          path: .github/actions/blocked-files-checkout
+
+      - name: Fetch ESLint evaluator
+        uses: actions/checkout@v2
+        with:
+          repository: betrybe/eslint-linter-action
+          ref: v3
+          token: ${{ secrets.GIT_HUB_PAT }}
+          path: .github/actions/eslint-evaluator
+
+      - name: Setup NodeJS
+        uses: actions/setup-node@v1.4.4
+        with:
+          node-version: '14'
+
+      - name: Restore protected files
+        uses: ./.github/actions/blocked-files-checkout
+        with:
+          restore_branch: 'master'
+
+      - name: Run ESLint evaluator
+        uses: ./.github/actions/eslint-evaluator
         with:
           token: ${{ secrets.GITHUB_TOKEN }}
+          pr_number: ${{ github.event.inputs.pr_number }}
 
-  evaluator_job:
-    name: Evaluator Job
-    runs-on: ubuntu-18.04
+  Evaluator:
+    runs-on: self-hosted
+    name: Evaluator
+    needs: [ESLint-Evaluator]
     services:
-      mysql:
-        image: mysql:8.0.21
-        env:
-          MYSQL_ROOT_PASSWORD: 'password'
-        ports:
-          - 3306:3306
-          - 33060:33060
-        options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=5s --health-retries=3
+       mysql:
+         image: mysql:8.0.21
+         env:
+           MYSQL_ROOT_PASSWORD: 'password'
+         ports:
+           - 3306:3306
+           - 33060:33060
+         options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=5s --health-retries=3
```

```suggestion
      mysql:
        image: mysql:8.0.21
        env:
          MYSQL_ROOT_PASSWORD: 'password'
        ports:
          - 3306:3306
          - 33060:33060
        options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=5s --health-retries=3
```
Eu reparei algo muito singelo aqui, não sei se é pertinente, mas já ouvi falar que arquivos YAML são bem sensíveis à indentação. No repositório de _staging_ eu testei com a indentação errada e tudo funcionou, você acha válido corrigir isso ou como esse código já foi testado é melhor não mexer?

---

Acho que faz sentido sim. Parece que isso não influencia no avaliador, mas é bom deixar padronizado :D 

Fiz essa mesma [alteração no template](https://github.com/betrybe/sd-0x-project-blogs-api/pull/28), pode dar o review lá, por favor?

---

CR feito!